scripts/extract.py

```diff
@@ -44,19 +45,33 @@ def process(self):
         else:
             self.extract_single_process()
 
-        self.alignments.write_alignments(self.faces.faces_detected)
-
+        self.write_alignments()
         images, faces = Utils.finalize(self.images.images_found,
                                        self.faces.num_faces_detected,
                                        self.faces.verify_output)
         self.images.images_found = images
         self.faces.num_faces_detected = faces
 
+    def write_alignments(self):
+        self.alignments.write_alignments(self.faces.faces_detected)
+
+
     def extract_single_process(self):
         """ Run extraction in a single process """
+
+        alignments = self.alignments.read_alignments()
```

Comment:
Why are you reading alignments here? It shouldn't exist at this point. If it's to continue off from where previously left off, then we have 'skip existing' for that.